test/e2e-helm/e2e_helm_metrics_test.go

```diff
@@ -0,0 +1,182 @@
+				filepath.Join(tc.Dir, "config", "default", "kustomization.yaml"),
+				"#- ../prometheus", "#")).To(Succeed())
+
+			By("Deploying project on the cluster")
+			err := tc.Make("deploy", "IMG="+tc.ImageName)
+			Expect(err).Should(Succeed())
+
+			By("Checking if the Operator project Pod is running")
+			verifyControllerUp := func() error {
+				By("Getting the controller-manager pod name")
+				podOutput, err := tc.Kubectl.Get(
+					true,
+					"pods", "-l", "control-plane=controller-manager",
+					"-o", "go-template={{ range .items }}{{ if not .metadata.deletionTimestamp }}{{ .metadata.name }}"+
+						"{{ \"\\n\" }}{{ end }}{{ end }}")
+				Expect(err).NotTo(HaveOccurred())
+
+				By("Ensuring the created controller-manager Pod")
+				podNames := kbtestutils.GetNonEmptyLines(podOutput)
+				if len(podNames) != 1 {
+					return fmt.Errorf("expect 1 controller pods running, but got %d", len(podNames))
+				}
+				controllerPodName = podNames[0]
+				Expect(controllerPodName).Should(ContainSubstring("controller-manager"))
+
+				By("Checking the controller-manager Pod is running")
+				status, err := tc.Kubectl.Get(
+					true,
+					"pods", controllerPodName, "-o", "jsonpath={.status.phase}")
+				Expect(err).NotTo(HaveOccurred())
+				if status != "Running" {
+					return fmt.Errorf("controller pod in %s status", status)
+				}
+				return nil
+			}
+			Eventually(verifyControllerUp, time.Minute, time.Second).Should(Succeed())
+
+			By("Ensuring the created ServiceMonitor for the manager")
+			_, err = tc.Kubectl.Get(
+				true,
+				"ServiceMonitor",
+				fmt.Sprintf("e2e-%s-controller-manager-metrics-monitor", tc.TestSuffix))
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Ensuring the created metrics Service for the manager")
+			_, err = tc.Kubectl.Get(
+				true,
+				"Service",
+				fmt.Sprintf("e2e-%s-controller-manager-metrics-service", tc.TestSuffix))
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Granting permissions to access the metrics and read the token")
+			_, err = tc.Kubectl.Command(
+				"create",
+				"clusterrolebinding",
+				fmt.Sprintf("metrics-%s", tc.TestSuffix),
+				fmt.Sprintf("--clusterrole=e2e-%s-metrics-reader", tc.TestSuffix),
+				fmt.Sprintf("--serviceaccount=%s:default", tc.Kubectl.Namespace))
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Getting the token")
+			b64Token, err := tc.Kubectl.Get(
+				true,
+				"secrets",
+				"-o=jsonpath={.items[0].data.token}")
+			Expect(err).NotTo(HaveOccurred())
+			token, err := base64.StdEncoding.DecodeString(strings.TrimSpace(b64Token))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(token).NotTo(HaveLen(0))
+
+			By("Pulling docker curl image")
+			pullCurlImage := exec.Command("docker", "pull", curlImage)
+			_, err = tc.Run(pullCurlImage)
+			Expect(err).NotTo(HaveOccurred())
+
+			if IsRunningOnKind() {
+				By("Loading the curl image into the Kind cluster")
+				err = tc.LoadImageToKindClusterWithName(curlImage)
+				Expect(err).Should(Succeed())
+			}
```

Comment:
Is this necessary? I figured the only images that would need to be loaded into `kind` would be the ones that are not pullable from a remote registry.